java/java.editor/src/org/netbeans/modules/editor/java/GoToSupport.java

```diff
@@ -381,6 +385,35 @@ public static Context resolveContext(CompilationInfo controller, Document doc, i
                             el = e;
                         }
                     }
+                    if (el != null && el.getKind() == ElementKind.METHOD) {
+                        for (Element peer : el.getEnclosingElement().getEnclosedElements()) {
+                            if (peer.getKind().name().contains("RECORD_COMPONENT")) {
+                                try {
+                                    Class<?> recordComponent = Class.forName("javax.lang.model.element.RecordComponentElement", true, VariableTree.class.getClassLoader());
+                                    Method getAccessor = recordComponent.getDeclaredMethod("getAccessor");
+                                    Method getRecordComponents = TypeElement.class.getDeclaredMethod("getRecordComponents");
```

Comment:
I wonder if the reflection should be separated into TreeShims. But this can wait.

Reply:
It would be good if we wait and fix this at later stage.